libs/@blockprotocol/type-system/crate/src/ontology/shared/all_of/repr.rs

```diff
@@ -9,11 +9,7 @@ use crate::{repr, EntityTypeReference, ParseAllOfError};
 #[serde(rename_all = "camelCase", deny_unknown_fields)]
 pub struct AllOf<T> {
     #[cfg_attr(target_arch = "wasm32", tsify(optional, type = "T[]"))]
-    #[serde(
-        rename = "allOf",
-        default = "Vec::new",
-        skip_serializing_if = "Vec::is_empty"
-    )]
+    #[serde(rename = "allOf", default, skip_serializing_if = "Vec::is_empty")]
```

Comment:
Removing the `default = "Vec::new"` causes a build error for me:

```rust
@blockprotocol/type-system:build: error[E0277]: the trait bound `entity_type::repr::EntityTypeReference: Default` is not satisfied
@blockprotocol/type-system:build:   --> src/ontology/entity_type/repr.rs:41:5
@blockprotocol/type-system:build:    |
@blockprotocol/type-system:build: 41 | /     #[serde(flatten)]
@blockprotocol/type-system:build: 42 | |     all_of: repr::AllOf<EntityTypeReference>,
@blockprotocol/type-system:build:    | |____________________________________________^ the trait `Default` is not implemented for `entity_type::repr::EntityTypeReference`
@blockprotocol/type-system:build:    |
@blockprotocol/type-system:build:    = help: the trait `Deserialize<'de>` is implemented for `all_of::repr::AllOf<T>`
@blockprotocol/type-system:build: note: required for `all_of::repr::AllOf<entity_type::repr::EntityTypeReference>` to implement `Deserialize<'_>`
```

Reply:
Serde's macro is not smart enough again, there is [no `T: Default` bound on `Vec`](https://doc.rust-lang.org/std/vec/struct.Vec.html#impl-Default-for-Vec%3CT,+Global%3E):
```
impl<T> Default for Vec<T> { ... }
```

Reverted in [`24e0a6b` (#1296)](https://github.com/blockprotocol/blockprotocol/pull/1296/commits/24e0a6bd459bbd15f15dfa2be549f28e40809de8)